model/task_lifecycle_test.go

```diff
@@ -3367,6 +3367,73 @@ func TestClearAndResetStaleStrandedHostTask(t *testing.T) {
 	assert.Equal("system", runningTask.Details.Type)
 }
 
+func TestClearAndResetStrandedHostTaskFailedOnly(t *testing.T) {
+	require.NoError(t, db.ClearCollections(host.Collection, task.Collection, task.OldCollection, build.Collection, VersionCollection))
+
+	dispTask := &task.Task{
+		Id:                      "dt",
+		Status:                  evergreen.TaskStarted,
+		Version:                 "version",
+		Activated:               true,
+		ActivatedTime:           time.Now(),
+		BuildId:                 "b",
+		ExecutionTasks:          []string{"et1", "et2"},
+		ResetFailedWhenFinished: true,
```

Comment:
I think you shouldn't set this here since the point of testing ClearAndResetStrandedHostTask is that it's supposed to do the setting.